### ISSUE TITLE
TextInput on focus - Maximum call stack size exceeded

### ISSUE THREAD

**Issue by Dellybro** [facebook/react-native#29596]
Please provide all the information requested. Issues that do not follow this format are likely to stall.

## Description

Upon focusing on a text field i get a "Maximum call stack  size exceeded", this only happens when __DEV__ = true

The call stack is calling "isValidArgument" an infinite amount of times, when i follow it the function seems only to run when __DEV__ is true.

## React Native version:
Run `react-native info` in your terminal and copy the results here.

```
System:
    OS: macOS 10.15.6
    CPU: (8) x64 Intel(R) Core(TM) i7-3770 CPU @ 3.40GHz
    Memory: 195.08 MB / 16.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 8.15.1 - /usr/local/bin/node
    Yarn: Not Found
    npm: 6.13.7 - /usr/local/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.9.3 - /Users/Dellybro/.rvm/gems/ruby-2.6.3/bin/pod
  SDKs:
    iOS SDK:
      Platforms: iOS 13.4, DriverKit 19.0, macOS 10.15, tvOS 13.4, watchOS 6.2
    Android SDK: Not Found
  IDEs:
    Android Studio: 3.5 AI-191.8026.42.35.5900203
    Xcode: 11.4.1/11E503a - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.2 - /usr/bin/javac
    Python: 2.7.16 - /usr/bin/python
  npmPackages:
    @react-native-community/cli: Not Found
    react: 16.13.1 => 16.13.1 
    react-native: 0.63.2 => 0.63.2 
  npmGlobalPackages:
    *react-native*: Not Found
```

## Steps To Reproduce
Provide a detailed list of steps that reproduce the issue.

I'm not really sure how to reproduce this error, because it doesn't happen on every input field. It randomly started happening, and idk what caused it.

## Expected Results

I expect that when __DEV__ is true and false, i should be able to focus on an text input

## Snack, code example, screenshot, or link to a repository:
Please provide a Snack (https://snack.expo.io/), a link to a repository on GitHub, or provide a minimal code example that reproduces the problem.
You may provide a screenshot of the application if you think it is relevant to your bug report.
Here are some tips for providing a minimal example: https://stackoverflow.com/help/mcve

```
[[33,45,71,71,71,71,71,71,71,54,71,71,71,71,71,71,71],[13,5,3,3,3,3,3,3,0,0,3,3,3,3,3,3,0],[[{"componentStack":null,"id":1,"originalMessage":"Maximum call stack size exceeded","extraData":{"suppressRedBox":true,"rawStack":"RangeError: Maximum call stack size exceeded\n    at isValidArgument (http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:2680:57)\n    at isValidArgument (http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:2700:52)\n    at isValidArgument (http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:2700:52)\n    at isValidArgument (http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:2700:52)\n    at isValidArgument (http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:2700:52)\n    at isValidArgument (http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:2700:52)\n    at isValidArgument (http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:2700:52)\n    at isValidArgument (http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:2700:52)\n    at isValidArgument (http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:2700:52)\n    at isValidArgument (http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:2700:52)"},"isFatal":true,"stack":[{"column":56,"methodName":"isValidArgument","file":"http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false","lineNumber":2680,"arguments":[]},{"column":51,"methodName":"isValidArgument","file":"http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false","lineNumber":2700,"arguments":[]},{"column":51,"methodName":"isValidArgument","file":"http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false","lineNumber":2700,"arguments":[]},{"column":51,"methodName":"isValidArgument","file":"http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false","lineNumber":2700,"arguments":[]},{"column":51,"methodName":"isValidArgument","file":"http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false","lineNumber":2700,"arguments":[]},{"column":51,"methodName":"isValidArgument","file":"http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false","lineNumber":2700,"arguments":[]},{"column":51,"methodName":"isValidArgument","file":"http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false","lineNumber":2700,"arguments":[]},{"column":51,"methodName":"isValidArgument","file":"http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false","lineNumber":2700,"arguments":[]},{"column":51,"methodName":"isValidArgument","file":"http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false","lineNumber":2700,"arguments":[]},{"column":51,"methodName":"isValidArgument","file":"http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false","lineNumber":2700,"arguments":[]}],"message":"RangeError: Maximum call stack size exceeded","name":"RangeError"}],["didSendNetworkData"],["didReceiveNetworkResponse"],["didReceiveNetworkData"],["didReceiveNetworkIncrementalData"],["didReceiveNetworkDataProgress"],["didCompleteNetworkResponse"],[{"responseType":"blob","url":"http://localhost:8081/symbolicate","data":{"trackingName":"unknown","string":"{\"stack\":[{\"file\":\"http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false\",\"methodName\":\"isValidArgument\",\"arguments\":[],\"lineNumber\":2680,\"column\":56},{\"file\":\"http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false\",\"methodName\":\"isValidArgument\",\"arguments\":
```

**Comment by co-de** [comment]
I experience the same thing. When I deploy me react-native app to my phone in "debug" mode I see this error. If I deploy the app in release mode the error disappears.

**react-native info**
```
info Fetching system and libraries information...
System:
    OS: macOS 10.15.6
    CPU: (4) x64 Intel(R) Core(TM) i5-4278U CPU @ 2.60GHz
    Memory: 24.65 MB / 8.00 GB
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 8.17.0 - /usr/local/bin/node
    Yarn: 1.15.2 - /usr/local/bin/yarn
    npm: 6.13.4 - /usr/local/bin/npm
    Watchman: Not Found
  Managers:
    CocoaPods: 1.9.3 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: iOS 13.6, DriverKit 19.0, macOS 10.15, tvOS 13.4, watchOS 6.2
    Android SDK:
      API Levels: 23, 26, 27, 28
      Build Tools: 28.0.3
      System Images: android-27 | Google Play Intel x86 Atom, android-28 | Google APIs Intel x86 Atom
      Android NDK: Not Found
  IDEs:
    Android Studio: 3.5 AI-191.8026.42.35.5791312
    Xcode: 11.6/11E708 - /usr/bin/xcodebuild
  Languages:
    Java: 1.8.0_131 - /usr/bin/javac
    Python: 2.7.16 - /usr/bin/python
  npmPackages:
    @react-native-community/cli: Not Found
    react: 16.13.1 => 16.13.1
    react-native: 0.63.2 => 0.63.2
  npmGlobalPackages:
    *react-native*: Not Found
```

**Comment by safaiyeh** [comment]
Hi @Dellybro  and @co-de it would be helpful to have a repro with steps to get the error to happen. Could you try to provide that? Thanks!

**Comment by Dellybro** [comment]
@safaiyeh the problem is i don't know how to reproduce it. Half of my app works, and then the other half does not. I'm doing nothing different, i import the input just like i do everywhere else.

**Comment by safaiyeh** [comment]
How about how you're using the component? Something to narrow it down. Code snippet of how it's used etc..

**Comment by Dellybro** [comment]
@safaiyeh 

![Screen Shot 2020-08-19 at 10 58 50 AM](https://user-images.githubusercontent.com/10732096/90672515-1569e300-e20b-11ea-84f5-6f8858035caa.png)


```
import React from 'react';
// Modules
import ApplyStyles from '@fyresite/apply-styles';
// Components
import { View, TextInput, Text } from 'react-native';
import Icon from '../Icon';
// Styles
import styles from './styles';
import { PlaceholderTextColor, danger } from '../../styles';

class StyledCustomInput extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            secureTextEntry: props.inputProps?.secureTextEntry ?? false,
        };
        this.wrappedInstance = React.createRef();
    }

    toggleSecureEntry() {
        this.setState({ secureTextEntry: !this.state.secureTextEntry });
    }

    render() {
        let {
            error,
            errorMessage,
            validator,
            style,
            label,
            inputStyles,
            labelStyles,
            iconProps,
            marginBottom = 0,
            inputProps = {},
            variant = 'bordered',
            ...rest
        } = this.props;

        let _styles = ApplyStyles(style, [{ apply: true, style: { marginBottom } }]);
        let _rootStyles = ApplyStyles(styles.root, [
            { apply: true, style: styles[variant] },
            { apply: error, style: styles.error },
        ]);
        let _inputStyles = ApplyStyles(styles.input, inputStyles, [
            {
                apply: !!iconProps,
                style: styles.hasIcon,
            },
        ]);
        let _labelStyles = ApplyStyles(styles.label, labelStyles, [{ apply: error, style: styles.error }]);
        // Default and error value
        if (iconProps) {
            if (!iconProps.style) iconProps.style = {};
            if (!iconProps.style.fontSize) iconProps.style.fontSize = 26;
            if (error) iconProps.style.color = danger;
        }

        return (
            <View style={_styles}>
                {label && <Text style={_labelStyles}>{label}</Text>}
                <View style={_rootStyles} {...rest}>
                    {iconProps && <Icon active {...iconProps} />}
                    <TextInput
                        ref={(el) => (this.wrappedInstance = el)}
                        placeholderTextColor={PlaceholderTextColor}
                        style={_inputStyles}
                        {...inputProps}
                        secureTextEntry={this.state.secureTextEntry}
                    />
                    {inputProps.secureTextEntry && (
                        <Icon
                            onPress={this.toggleSecureEntry.bind(this)}
                            active
                            {...iconProps}
                            type="MaterialIcons"
                            name={this.state.secureTextEntry ? 'visibility-off' : 'visibility'}
                        />
                    )}
                </View>
            </View>
        );
    }
}

export default StyledCustomInput;

```

**Comment by Dellybro** [comment]
@safaiyeh so the crash is coming from 

react-native/Libraries/BatchedBridge 
react-native-windows/Libraries/BatchedBridge 

The function "isValidArgument" is re-rendering and infinite amount of times

**Comment by safaiyeh** [comment]
Thanks for the investigation! I'll tag this appropriately

**Comment by Dellybro** [comment]
@safaiyeh The app i'm working on is client related, so unfortunately i can't give full source code. But i'd always be willing to dive deeper, and even do some sort of zoom call to help debug this issue if something like that was possible, because at this current moment, i am unable to actually use these inputs, but luckily the app is 99% finished. This started happening after upgrading all the packages from 0.59 to 0.63 

This project was put on the backburner for a year, so i actually reinitialized a new react app and copy and pasted the code into the new repo.

**Comment by marinashe** [comment]
I had the exact same problem.
But the problem was not in TextInput, but in a component that  wrapped TextInput. For me it was InputScrollView from library react-native-input-scroll-view. In this library it was fixed by this commit https://github.com/baijunjie/react-native-input-scroll-view/commit/f4361eb871d70cd947fced7fa3189c81d05458b3

**Comment by Bro-GunHo** [comment]
same problem.
only debug mode

**Comment by Dellybro** [comment]
I've got more information about this. 

This is happening when the keyboard is toggled, after untoggling the keyboard this no longer happens. And as i said this is not on all fields, only a select amount of fields and it's the same fields. But the weird part is, all the pages have the exact same stuff, keyboard aware view, and text input.

**Comment by Meycon** [comment]
Is there no solution?

**Comment by Meycon** [comment]
I had the exact same problem.
The fix might just require find & replace all instances to currentlyFocusedField

`  handleKeyboardDidShow = (event) => {
    if (this.props.disabled) return
    const { height: windowHeight } = Dimensions.get('window')
    const keyboardHeight = event.endCoordinates.height
    const currentlyFocusedField = TextInputState.currentlyFocusedField()
    if (currentlyFocusedField !== null)
      UIManager.measure(
        currentlyFocusedField,
        (originX, originY, width, height, pageX, pageY) => {
          const fieldHeight = height
          const fieldTop = pageY
          const androidBonus = Platform.OS === 'android' ? scale(30) : scale(10)
          const gap =
            windowHeight -
            keyboardHeight -
            (fieldTop + fieldHeight + androidBonus)
          if (gap >= 0) {
            return
          }
          Animated.timing(this.state.shift, {
            toValue: gap,
            duration: 150,
            useNativeDriver: true,
          }).start()
        }
      )
  }`

**Comment by slothwerks-studio** [comment]
Hey guys... I just updated from React Native v0.61.5 to v0.63.4.  Initially, the upgrade (using upgrade-patch) seemed fine.  But then I began meticulously updating the code for Android found here:

https://react-native-community.github.io/upgrade-helper/?from=0.61.5&to=0.63.4

After doing this, I encountered the issue you describe (maximum call stack exceeded).

Seeing that someone referenced folly-convertible, I decided to roll back the Flipper-related code.  This resolved the issue on my end.  This was in three files:

- gradle.properties
- MainApplication.java
- build.gradle

I then wiped out the new ReactNativeFlipper.java as well.

This didn't take too terribly long, but ultimately I'll likely need to rebuild the project using the latest versions of React Native and React... something I don't look forward to.
